File: works/tmp/10997.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;
import java.util.Collections;

public class Main
{
    static int N;

    static void Draw(int r, int c, int cnt, int depth, char[][] stars)
    {
        stars[r][c] = '*';
        // up
        for(int i = 0; i < cnt-1; i++)
        {
            r--;
            stars[r][c] = '*';
        }

        // right
        for(int i = 0; i < cnt-1; i++)
        {
            c++;
            if(c >= stars[r].length) break;
            stars[r][c] = '*';
        }

        if(depth == N-1) return;

        // down
        for(int i = 0; i < cnt+2-1; i++)
        {
            r++;
            stars[r][c] = '*';
        }

        // left
        for(int i = 0; i < cnt+2-1; i++)
        {
            c--;
            stars[r][c] = '*';
        }

        Draw(r, c, cnt+4, depth+1, stars);

    }

    static void Print(char[][] stars)
    {
        for(int i = 0; i < stars.length; i++)
        {
            if(i == 1)
                System.out.println('*');
            else
            {
                for(int j = 0; j < stars[i].length; j++)
                    System.out.print(stars[i][j]);
                System.out.println();
            }
        }
    }

    public static void main(String[] args) throws IOException
    {
        Scanner sc = new Scanner(System.in);
        N = sc.nextInt();
        if(N == 1) { System.out.print('*'); return; }

        char[][] stars = new char[4*N-2+1][4*N-4+1];
        for(int i = 0; i < stars.length; i++)
        {
            for(int j = 0; j < stars[0].length; j++)
            {
                stars[i][j] = ' ';
            }
        }

        Draw(2*N, 2*N-2, 3, 0, stars);
        
        Print(stars);

    }
}

File: works/tmp/10997.kt
import java.util.*

fun Draw(_r:Int, _c:Int, cnt:Int, depth:Int, stars:Array<CharArray>, N:Int)
{
    var r = _r
    var c = _c

    stars[r][c] = '*'
    for(i in 0 until cnt-1)
    {
        r--
        stars[r][c] = '*'
    }
    for(i in 0 until cnt-1)
    {
        c++
        if(c >= stars[r].size) break
        stars[r][c] = '*'
    }

    if(depth == N-1) return

    for(i in 0 until cnt+2-1)
    {
        r++
        stars[r][c] = '*'
    }
    for(i in 0 until cnt+2-1)
    {
        c--
        stars[r][c] = '*'
    }

    Draw(r, c, cnt+4, depth+1, stars, N)
}

fun Print(stars:Array<CharArray>)
{
    for(i in 0 until stars.size)
    {
        if(i == 1)
            println('*')
        else
        {
            for(j in 0 until stars[i].size)
                print(stars[i][j])
            println()
        }
    }
}

fun main()
{
    val sc = Scanner(System.`in`)
    val N = sc.nextInt()
    if(N==1)
    {
        print('*')
        return
    }

    val stars = Array(4*N-2+1) { CharArray(4*N-4+1) {' '} }

    Draw(2*N, 2*N-2, 3, 0, stars, N)

    Print(stars)
}
